File: public/examples/cpp/basic/03_collections.cpp
#include <iostream>
#include <vector>
#include <map>

int main() {
    // Vector (Dynamic array)
    std::vector<std::string> fruits = {"apple", "banana"};
    fruits.push_back("cherry");
    
    std::cout << "Fruits:" << std::endl;
    for (const auto& fruit : fruits) {
        std::cout << "- " << fruit << std::endl;
    }

    // Map (Dictionary)
    std::map<std::string, int> ages;
    ages["Alice"] = 25;
    ages["Bob"] = 30;
    
    std::cout << "Alice's age: " << ages["Alice"] << std::endl;
    
    return 0;
}

File: public/examples/go/basic/03_collections.go
package main

import "fmt"

func main() {
    // Slices (dynamic arrays)
    fruits := []string{"apple", "banana"}
    fruits = append(fruits, "cherry")
    fmt.Println("Slices:", fruits)

    // Maps (Key-Value)
    ages := map[string]int{
        "Alice": 25,
        "Bob":   30,
    }
    fmt.Println("Alice's age:", ages["Alice"])

    // Iterating over collections
    for i, fruit := range fruits {
        fmt.Printf("Index %d: %s\n", i, fruit)
    }
}
